src/components/network/geo-data.js

```diff
@@ -88,4 +91,45 @@ export default class GeoData {
 
         return [ substationPosition1, substationPosition2 ];
     };
+
+    getCoordinateInLine(positions, lineDistance, percent) {
+        if (percent > 100 || percent < 0) {
+            throw Error("percent value incorrect: " + percent);
+        }
+        if (lineDistance <= 0) {
+            return null
+        }
+
+        let wantedDistance = lineDistance * percent / 100;
```

Comment:
I think these distances are already computed somewhere for the arrow layers, maybe we can share the code ?